In [550]:
#Import libraries
import numpy as np
import pandas as pd
from pandas import Series, DataFrame

#Import datasets (movieds and movie2ds)
movie_ds1= pd.read_csv('movieds.csv')
movie_ds2= pd.read_csv('movie2ds.csv')

#print the shape of datasets to confirm they are correctly imported
print(movie_ds1.shape)
print(movie_ds2.shape)


(15069, 10)
(15069, 6)


In [ ]:
Exploring the Datasets

In [552]:
#View the DataFrames
print(movie_ds1.head())
movie_ds2.head()

                                               Title  Year  Age  IMDb  \
0                                       Trigger Pals  1939  NaN   5.3   
1         One Way Astronaut: The Mars One Initiative  2017  NaN   NaN   
2                                         Open Grave  2013  18+   6.2   
3  Metallica: Kill 'Em All to St. Anger - The Ult...  2006  NaN   NaN   
4                                    Beyond the Pale  2014  NaN   7.4   

  Rotten Tomatoes              Directors                    Genres  \
0             NaN           Sam Newfield  Action,Adventure,Western   
1             NaN                    NaN                       NaN   
2             18%  Gonzalo López-Gallego   Horror,Mystery,Thriller   
3             NaN                    NaN                       NaN   
4             NaN            Ja-Ann Wang               Short,Drama   

                 Country Language  Runtime  
0          United States  English     59.0  
1                    NaN      NaN      NaN  
2  Un

ID                                          Title  Netflix  Hulu  \
0  13825  The Jungle Book: Search for the Lost Treasure        0     0   
1  15943           Vellaiya Irukiravan Poi Solla Maatan        0     0   
2   6233                                  Hollows Grove        0     0   
3  16194                              Web of the Spider        0     0   
4   7870                                 Life in Danger        0     0   

   Prime Video  Disney+  
0            1        0  
1            1        0  
2            1        0  
3            1        0  
4            1        0

In [526]:
#view the metadata of datasest
print(movie_ds1.info())
print(movie_ds2.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15069 entries, 0 to 15068
Data columns (total 10 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Title            15069 non-null  object 
 1   Year             15069 non-null  int64  
 2   Age              6609 non-null   object 
 3   IMDb             14543 non-null  float64
 4   Rotten Tomatoes  4636 non-null   object 
 5   Directors        14396 non-null  object 
 6   Genres           14814 non-null  object 
 7   Country          14672 non-null  object 
 8   Language         14507 non-null  object 
 9   Runtime          14533 non-null  float64
dtypes: float64(2), int64(1), object(7)
memory usage: 1.1+ MB
None
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15069 entries, 0 to 15068
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   ID           15069 non-null  int64 
 1   Title        15069 non-null  object


In [528]:
#compare the column names
print(movie_ds1.columns)
print(movie_ds2.columns)

Index(['Title', 'Year', 'Age', 'IMDb', 'Rotten Tomatoes', 'Directors',
       'Genres', 'Country', 'Language', 'Runtime'],
      dtype='object')
Index(['ID', 'Title', 'Netflix', 'Hulu', 'Prime Video', 'Disney+'], dtype='object')


Data Cleaning and Modeling:

In [554]:
#Create a new DataFrame Consisting these columns: ID,Title,Age,Genres,Langauge,IMDb
movie_nds= movie_ds1.loc[:, ['Title','Age','Genres','Language', 'IMDb', 'Rotten Tomatoes']]

#View the new DataFrame 
movie_nds.head()



Title  Age  \
0                                       Trigger Pals  NaN   
1         One Way Astronaut: The Mars One Initiative  NaN   
2                                         Open Grave  18+   
3  Metallica: Kill 'Em All to St. Anger - The Ult...  NaN   
4                                    Beyond the Pale  NaN   

                     Genres Language  IMDb Rotten Tomatoes  
0  Action,Adventure,Western  English   5.3             NaN  
1                       NaN      NaN   NaN             NaN  
2   Horror,Mystery,Thriller  English   6.2             18%  
3                       NaN      NaN   NaN             NaN  
4               Short,Drama  English   7.4             NaN

In [532]:
movie_nds.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15069 entries, 0 to 15068
Data columns (total 6 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Title            15069 non-null  object 
 1   Age              6609 non-null   object 
 2   Genres           14814 non-null  object 
 3   Language         14507 non-null  object 
 4   IMDb             14543 non-null  float64
 5   Rotten Tomatoes  4636 non-null   object 
dtypes: float64(1), object(5)
memory usage: 706.5+ KB


In [438]:
#Check for the missing data
movie_nds.isnull().sum()

Title                  0
Age                 8460
Genres               255
Language             562
IMDb                 526
Rotten Tomatoes    10433
dtype: int64

In [556]:
#Combine both datasets : movie_nds and movie_ds2 using merge() function and Title as Primary Key
movie_fds = pd.merge(movie_nds , movie_ds2,
                    on = 'Title')
#print the new dataframe
movie_fds.head()


Title  Age  \
0                                       Trigger Pals  NaN   
1                                         Open Grave  18+   
2  Metallica: Kill 'Em All to St. Anger - The Ult...  NaN   
3                                    Beyond the Pale  NaN   
4                                        White Tiger  13+   

                         Genres        Language  IMDb Rotten Tomatoes     ID  \
0      Action,Adventure,Western         English   5.3             NaN  15592   
1       Horror,Mystery,Thriller         English   6.2             18%   9730   
2                           NaN             NaN   NaN             NaN   8681   
3                   Short,Drama         English   7.4             NaN   1987   
4  Action,Adventure,Fantasy,War  Russian,German   5.0             NaN  16377   

   Netflix  Hulu  Prime Video  Disney+  
0        0     0            1        0  
1        0     0            1        0  
2        0     0            1        0  
3        0     0            1        0  
4        0     0            1        0

In [536]:
movie_fds.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13423 entries, 0 to 13422
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Title            13423 non-null  object 
 1   Age              5886 non-null   object 
 2   Genres           13206 non-null  object 
 3   Language         12929 non-null  object 
 4   IMDb             12953 non-null  float64
 5   Rotten Tomatoes  4121 non-null   object 
 6   ID               13423 non-null  int64  
 7   Netflix          13423 non-null  int64  
 8   Hulu             13423 non-null  int64  
 9   Prime Video      13423 non-null  int64  
 10  Disney+          13423 non-null  int64  
dtypes: float64(1), int64(5), object(5)
memory usage: 1.1+ MB


In [558]:
#Find and Deleting duplicates
clean_m_fds= movie_fds.drop_duplicates(subset = ('ID'))
#Delete Null values from IMDb
clean_m_fds = movie_fds.dropna(subset = ["IMDb"] )
#Print the shape
clean_m_fds.shape

(12953, 11)

In [560]:
#Combine both datasets : movie_nds and movie_ds2 using merge() function and Title as Primary Key
movie_fds = pd.merge(movie_nds , movie_ds2,
                    on = 'Title')
#print the new dataframe
movie_fds.head()


Title  Age  \
0                                       Trigger Pals  NaN   
1                                         Open Grave  18+   
2  Metallica: Kill 'Em All to St. Anger - The Ult...  NaN   
3                                    Beyond the Pale  NaN   
4                                        White Tiger  13+   

                         Genres        Language  IMDb Rotten Tomatoes     ID  \
0      Action,Adventure,Western         English   5.3             NaN  15592   
1       Horror,Mystery,Thriller         English   6.2             18%   9730   
2                           NaN             NaN   NaN             NaN   8681   
3                   Short,Drama         English   7.4             NaN   1987   
4  Action,Adventure,Fantasy,War  Russian,German   5.0             NaN  16377   

   Netflix  Hulu  Prime Video  Disney+  
0        0     0            1        0  
1        0     0            1        0  
2        0     0            1        0  
3        0     0            1        0  
4        0     0            1        0

In [510]:
clean_m_fds.shape

(12953, 11)

Apparently there are no duplicates.

In [566]:
#Find French language movies and drop the rest of rows that doesnt contain 'French' as a 'Language'
fmovie = movie_fds[movie_fds['Language'].str.contains(pat= 'French' , na = False)]
#print the result
fmovie.head()


Title  Age                Genres  \
27        Mystery of the Pink Dolphin  NaN           Documentary   
44                      My Revolution  NaN                 Drama   
48   The Story of the Twelve Apostles  NaN     Documentary,Short   
70                  Girl on a Bicycle  18+  Comedy,Drama,Romance   
100                            Razzia  NaN     Drama,History,War   

                            Language  IMDb Rotten Tomatoes     ID  Netflix  \
27   French,English,Portuguese,Dutch   7.1             NaN   9219        0   
44                            French   3.5             NaN   9193        0   
48            English,French,Spanish   6.6             NaN  14854        0   
70     English,French,German,Italian   6.2              8%   5524        0   
100            French,German,Yiddish   6.2             44%  10555        0   

     Hulu  Prime Video  Disney+  
27      0            1        0  
44      0            1        0  
48      0            1        0  
70      0            1        0  
100     0            1        0

In [564]:
fmovie.shape


(638, 11)

There are 638 rwos with French language movies 

In [568]:
#Rearranging the columns
fmovie_table=fmovie [['ID' , 'Title', 'Language' , 'Genres' , 'IMDb' , 'Rotten Tomatoes' , 'Age' , 'Netflix', 'Hulu','Prime Video','Disney+']]
#print the result
print(fmovie_table)


          ID                             Title  \
27      9219       Mystery of the Pink Dolphin   
44      9193                     My Revolution   
48     14854  The Story of the Twelve Apostles   
70      5524                 Girl on a Bicycle   
100    10555                            Razzia   
...      ...                               ...   
13387   8299                    Made in France   
13400   1089                    Among Dead Men   
13407  14816          The Special Relationship   
13414  13181                       The Decline   
13418  13673           The High Cost of Living   

                              Language                             Genres  \
27     French,English,Portuguese,Dutch                        Documentary   
44                              French                              Drama   
48              English,French,Spanish                  Documentary,Short   
70       English,French,German,Italian               Comedy,Drama,Romance   
100           

In [573]:
#Print Descriptive Statistics
print(fmovie.describe())

             IMDb            ID     Netflix        Hulu  Prime Video  \
count  626.000000    638.000000  638.000000  638.000000   638.000000   
mean     6.373482   8740.227273    0.253918    0.087774     0.677116   
std      0.990014   4886.371064    0.435593    0.283188     0.467946   
min      2.900000      1.000000    0.000000    0.000000     0.000000   
25%      5.825000   4440.250000    0.000000    0.000000     0.000000   
50%      6.500000   9149.500000    0.000000    0.000000     1.000000   
75%      7.100000  13297.250000    1.000000    0.000000     1.000000   
max      8.800000  16741.000000    1.000000    1.000000     1.000000   

          Disney+  
count  638.000000  
mean     0.029781  
std      0.170115  
min      0.000000  
25%      0.000000  
50%      0.000000  
75%      0.000000  
max      1.000000  


Getting an idea about how the data are looking 

In [575]:
#At this stage save the new dataset for future use and reference
fmovie_table.to_csv('French_movies.csv')


In [577]:
#Sorting the Data By highest Rating :IMDb and Rotten Tomato (Descending)
hr_fmovie=fmovie.sort_values(['IMDb', 'Rotten Tomatoes'],ascending= False)
#Print the top 20 French Language movies sorted by highest rating based on IMDb and Rotten Tomatoes
hr_fmovie.head(20)

Title  Age  \
790                          Inception  13+   
827              It's a Wonderful Life   7+   
6769                    The Green Mile  18+   
1721                              Home  NaN   
2590        Inner Worlds, Outer Worlds  NaN   
2524                     The Lion King  all   
10895                 Django Unchained  18+   
1978              Inglourious Basterds  18+   
2567                           Coexist  NaN   
9333                The Weekend Sailor  NaN   
1995        Portrait of a Lady on Fire  18+   
2394   Monty Python and the Holy Grail   7+   
9851                             Queen  13+   
6717                   Flakes & Flames  NaN   
11370                       Skydancers  NaN   
5749                 Kill Bill: Vol. 1  18+   
3735                   The Incredibles   7+   
5980                     Groundhog Day   7+   
6859                        Diabolique  NaN   
9100                       Ratatouille  all   

                                                 Genres  \
790                    Action,Adventure,Sci-Fi,Thriller   
827                                Drama,Family,Fantasy   
6769                        Crime,Drama,Fantasy,Mystery   
1721   Animation,Adventure,Comedy,Family,Fantasy,Sci-Fi   
2590                                Documentary,History   
2524           Animation,Adventure,Drama,Family,Musical   
10895                                     Drama,Western   
1978                                Adventure,Drama,War   
2567                       Documentary,History,News,War   
9333                                        Documentary   
1995                                      Drama,Romance   
2394                           Adventure,Comedy,Fantasy   
9851                             Adventure,Comedy,Drama   
6717                                Documentary,History   
11370                                       Documentary   
5749                              Action,Crime,Thriller   
3735                  Animation,Action,Adventure,Family   
5980                             Comedy,Fantasy,Romance   
6859                Crime,Drama,Horror,Mystery,Thriller   
9100    Animation,Adventure,Comedy,Drama,Family,Fantasy   

                                         Language  IMDb Rotten Tomatoes  \
790                       English,Japanese,French   8.8             87%   
827                                English,French   8.6             94%   
6769                               English,French   8.6             78%   
1721                               English,French   8.6             NaN   
2590   English,French,Spanish,German,Hindi,Polish   8.6             NaN   
2524            English,Xhosa,Zulu,French,Spanish   8.5             93%   
10895               English,German,French,Italian   8.4             87%   
1978                English,German,French,Italian   8.3             89%   
2567                   Kinyarwanda,English,French   8.3             NaN   
9333            English,Spanish,French,Portuguese   8.3             NaN   
1995                               French,Italian   8.2             98%   
2394                         English,French,Latin   8.2             97%   
9851          Hindi,English,French,Japanese,Dutch   8.2             86%   
6717                English,French,German,Spanish   8.2             NaN   
11370                              English,French   8.2             NaN   
5749                      English,Japanese,French   8.1             85%   
3735                               English,French   8.0             97%   
5980                       English,French,Italian   8.0             96%   
6859                               French,English   8.0             96%   
9100                               English,French   8.0             96%   

          ID  Netflix  Hulu  Prime Video  Disney+  
790     6734        1     0            0        0  
827     6933        0     0            1        0  
6769   13612        0     1            0        0  
1721    6258        0

Above we have the Top 20 Movies.
Now we are going to find the least tapped movies, for this matter I will sort the movies based on the lowest rating.


In [579]:
#Save the Dataset as a CSV file(high rated French language movies)
hr_fmovie.to_csv('High Rated French language movie.csv')

In [581]:
#List and sort the bottom 20 French movies.(Ascending) 
lr_fmovie=fmovie.sort_values(['IMDb','Rotten Tomatoes'])
#print the outcome
lr_fmovie.head(20)

Title  Age  \
2331      Anneliese: The Exorcist Tapes  18+   
8025                   My Little Sister  NaN   
10712                  Hold Your Breath  18+   
2591                      The Summoning  NaN   
44                        My Revolution  NaN   
6577   The Boy with the Sun in His Eyes  NaN   
13253                Doom: Annihilation  18+   
1553                          Seclusion  18+   
8999                  Burn it Up Djassa  NaN   
932                      Hoboken Hollow  NaN   
1697           Paranormal Investigation  NaN   
11300              Birds of a Feather 2  NaN   
12753                       Final Draft  NaN   
7422                   Girls with Balls  NaN   
2677                   Days of Darkness  18+   
2775                  On Hostile Ground  NaN   
13400                    Among Dead Men  18+   
2288                         The Cradle  18+   
3716                                  3  NaN   
3929      The Amazing Praybeyt Benjamin  NaN   

                                        Genres  \
2331                                    Horror   
8025                                     Drama   
10712            Drama,Mystery,Sci-Fi,Thriller   
2591    Drama,Fantasy,Mystery,Romance,Thriller   
44                                       Drama   
6577                     Comedy,Drama,Thriller   
13253  Action,Adventure,Horror,Sci-Fi,Thriller   
1553                             Drama,Romance   
8999                         Crime,Drama,Music   
932                                     Horror   
1697                           Horror,Thriller   
11300                                   Comedy   
12753                          Horror,Thriller   
7422                             Comedy,Horror   
2677                      Comedy,Drama,Fantasy   
2775           Action,Adventure,Drama,Thriller   
13400        Action,Crime,Horror,Mystery,Sport   
2288                           Horror,Thriller   
3716                   Action,Adventure,Sci-Fi   
3929             Action,Comedy,Fantasy,Romance   

                              Language  IMDb Rotten Tomatoes     ID  Netflix  \
2331             English,German,French   2.9             NaN   1220        0   
8025             German,French,English   3.1             NaN   9168        0   
10712                   French,English   3.2             NaN   6220        0   
2591                    English,French   3.3             NaN  14877        0   
44                              French   3.5             NaN   9193        0   
6577             English,French,German   3.5             NaN  12889        0   
13253                   English,French   3.6             50%   4215        1   
1553                            French   3.6             NaN  11252        0   
8999                      French,Nushi   3.6             NaN   2622        0   
932                     English,French   3.7             NaN   6215        0   
1697                            French   3.7             NaN   9918        1   
11300                           French   3.7             NaN   2101        0   
12753                   English,French   3.7             NaN   4910        0   
7422                            French   3.8             38%   5535        1   
2677                    French,English   3.9             20%   3726        0   
2775                    English,French   4.0             NaN   9653        0   
13400           English,French,Spanish   4.0             NaN   1089        0   
2288                    English,French   4.1             NaN  13094        0   
3716                    English,French   4.1             NaN      9        0   
3929   Filipino,Tagalog,English,French   4.1             NaN  12622        1   

       Hulu  Prime Video  Disney+  
2331      0            1        0  
8025      0            1        0  
10712     0            1        0  
2591      0            1        0  
44        0            1        0  
6577      0            1        0  
13253     0            0        0  
1553 

In [583]:
#save the Dataset as a CSV file(least rated French Language Genres)
lr_fmovie.to_csv('Least Tapped French Language Genres.csv')

Now its time to check which Genre has the lowest ranking.
Horror and Thriller have the most repeat in the lowest 20 range of rating.

In [585]:
#Split the Genres values in French movies
genres=fmovie['Genres'].dropna().str.split(',')

#Count and print the values for Genres 
print(genres.value_counts())

Genres
[Drama]                                          63
[Documentary]                                    49
[Drama, Romance]                                 29
[Comedy, Drama]                                  29
[Comedy]                                         20
                                                 ..
[Adventure, Mystery, Romance]                     1
[Action, Adventure, Horror, Sci-Fi, Thriller]     1
[Action, Biography, Drama, History, War]          1
[Mystery, Romance, Thriller]                      1
[Action, Crime, Horror, Mystery, Sport]           1
Name: count, Length: 252, dtype: int64


In [587]:
#Convert genres column values to row for counting the correct repetition of the values
genres= genres.explode()
#Count and print the values for Genres 
print(genres.value_counts())

Genres
Drama          386
Comedy         147
Thriller       147
Romance        135
Action          92
Documentary     86
Adventure       85
Crime           76
Mystery         65
History         56
Family          52
Biography       50
Fantasy         49
War             48
Horror          39
Animation       35
Sci-Fi          30
Music           25
Musical         16
Sport           10
Western          9
Short            6
News             4
Name: count, dtype: int64


In [593]:
#Determine the average of each genres based on rating
#Call Genres and IMDb columns and drop the rest of the columns
genres_rating = fmovie[['Genres','IMDb']].dropna()

#Use Str.split() function to split the values in the column
genres_rating ['Genres']= genres_rating['Genres'].str.split(',')

#convert genres column values to row
genres_rating = genres_rating.explode('Genres')

#Use Groupby() function to Genres column values  distinctively
avg_g_r = genres_rating.groupby('Genres')

#calculate mean for IMDb rating and sort them Descending
avg_g_r = avg_g_r['IMDb'].mean().sort_values(ascending = False)
#print the outcome
print(avg_g_r)

Genres
News           7.125000
Animation      6.985714
Documentary    6.973077
Short          6.866667
Biography      6.846939
History        6.798148
Family         6.769231
Music          6.680000
Western        6.622222
War            6.604167
Fantasy        6.561224
Adventure      6.545882
Musical        6.506250
Sport          6.470000
Romance        6.388060
Drama          6.345669
Comedy         6.338095
Crime          6.269737
Mystery        6.265625
Sci-Fi         6.260000
Action         6.148913
Thriller       6.145517
Horror         5.576923
Name: IMDb, dtype: float64


According to the average IMDb rating for genres News and Animation and Documentary have the highest rate of rating.
Horror , Thriller and Action have the lowest rating.

Checking the most frequent Age for French movie.We can see most of the French movies are for the 18+ years old, so to make the Frecnh language movie available for various age group we need to add more under 18 years old and family movies.

In [595]:
#Use value_count() function to get the age distribution
age =  fmovie['Age'].value_counts()
#Print the outcome
print(age)

Age
18+    146
13+     63
7+      57
all     21
16+     13
Name: count, dtype: int64


'18+' has repeated the most (146 times) and '16+' the least.

In [601]:
#print the bottom 20 most repeated Age based on rating
print('Bottom 20 Most Repeated Age')
print(lr_fmovie.head(20).Age.mode())

#print the top 20 most repeated Age
print('Top 20 Most Repeated Age')
print(hr_fmovie.head(20).Age.mode())

#print the most repeated Age
print('Most Repeated Age')
print(fmovie.Age.mode())

Bottom 20 Most Repeated Age
0    18+
Name: Age, dtype: object
Top 20 Most Repeated Age
0    18+
Name: Age, dtype: object
Most Repeated Age
0    18+
Name: Age, dtype: object


In [605]:
#sorting the dataset on Age and  Rating
age_fmovie=fmovie.sort_values(['Age', 'IMDb','Rotten Tomatoes'],ascending= False)
#print the outcome
age_fmovie.head(20)

Title  Age  \
2524                                      The Lion King  all   
9100                                        Ratatouille  all   
12729                        The Greasy Hands Preachers  all   
1269                             BPM (Beats per Minute)  all   
7539   Bon Voyage, Charlie Brown (and Don't Come Back!)  all   
889                           The Great Mouse Detective  all   
10180                         The Princess and the Frog  all   
7078                          Santa and the Three Bears  all   
1649                                      The Moustache  all   
9990                              Swim Little Fish Swim  all   
8352                                      The Milky Way  all   
9577                                   Asterix the Gaul  all   
7384                                 The House of Magic  all   
4961                                           CodeGirl  all   
4027             The Brave Little Toaster to the Rescue  all   
511                                 Nicholas on Holiday  all   
11395                                            Cars 2  all   
1806                       Babar: King of the Elephants  all   
736                                           Anastasia  all   
12483                         How to Build a Better Boy  all   

                                                  Genres  \
2524            Animation,Adventure,Drama,Family,Musical   
9100     Animation,Adventure,Comedy,Drama,Family,Fantasy   
12729                                        Documentary   
1269                                               Drama   
7539                   Animation,Adventure,Comedy,Family   
889           Animation,Adventure,Family,Musical,Mystery   
10180  Animation,Adventure,Comedy,Family,Fantasy,Musi...   
7078                            Animation,Family,Fantasy   
1649                                       Drama,Mystery   
9990                                  Comedy,Drama,Music   
8352                                        Comedy,Drama   
9577    Animation,Action,Adventure,Comedy,Family,Fantasy   
7384           Animation,Adventure,Comedy,Family,Fantasy   
4961                                         Documentary   
4027                  Animation,Adventure,Family,Fantasy   
511                                        Comedy,Family   
11395            Animation,Adventure,Comedy,Family,Sport   
1806                                    Family,Animation   
736    Animation,Adventure,Drama,Family,Fantasy,Music...   
12483                               Comedy,Family,Sci-Fi   

                                              Language  IMDb Rotten Tomatoes  \
2524                 English,Xhosa,Zulu,French,Spanish   8.5             93%   
9100                                    English,French   8.0             96%   
12729                          English,French,Japanese   7.8             NaN   
1269                                            French   7.4             98%   
7539                                    English,French   7.3             NaN   
889                      English,French,German,Italian   7.2             79%   
10180                                   English,French   7.1             85%   
7078                                    English,French   7.0             NaN   
1649                          French,English,Cantonese   6.7             86%   
9990                                    English,French   6.6             55%   
8352                      French,Italian,Latin,Spanish   6.6             NaN   
9577                                            French   6.6             NaN   
7384                             English,French,German   6.3             73%   
4961                 English,Portuguese,Spanish,French   6.3             60%   
4027   Chinese,Mandarin,Japanese,Korean,French,English   6.3             40%   
511                                             French   6.2             NaN   
11395                  English,Japanese,Italian,French   6.1             39%   
1806

Family/Animation/Comedy Genres has the good ranking and availability for everyone.

In [324]:
#Save the dataset for future reference
age_fmovie.to_csv('Various Age Availability(High Rating).csv')